website/src/components/Chat/HideAllChatButton.tsx

```diff
@@ -0,0 +1,29 @@
+import { EyeOff } from "lucide-react";
+import { useTranslation } from "next-i18next";
+import React, { useCallback } from "react";
+import useSWRMutation from "swr/mutation";
+import { API_ROUTES } from "src/lib/routes";
+import { put } from "src/lib/api";
+
+import { ChatListItemIconButton } from "./ChatListItem";
+
+export const HideAllChatButton = ({
+  chatIds,
+  onHide,
+}: {
+  chatIds: string[];
+  onHide?: (params: { chatId: string }) => void;
+}) => {
+  const { trigger: triggerHide } = useSWRMutation(API_ROUTES.UPDATE_CHAT(), put);
+
+  const onClick = useCallback(async () => {
+    for (const chatId of chatIds) {
+      await triggerHide({ chat_id: chatId, hidden: true });
+      onHide?.({ chatId });
+    }
+  }, [onHide, triggerHide, chatIds]);
```

Comment:
I am not sure if this is how we want to do it, this is the easiest, but it produces a lot of traffic.

Reply:
Gotcha, I will try to work on creating another trigger, if that is possible. Will update you soon.

Reply:
It seems like creating a new trigger is the way to go, but I have no idea how to create another trigger/api endpoint. Would you be able to direct me a bit?

Reply:
@CertifiedJoon I have created #3423 to add the necessary backend endpoint for this

Reply:
@olliestanley Thank you:)

Reply:
@olliestanley I am so sorry to bother you again, but i've tried PUT to /api/chat/hide_all, and got a 404. Which url do i have to try?

Reply:
Pretty sure it should be `/chats` not `/chat`